### ISSUE TITLE
Store path of CLDF metadata relative to pylingdocs.cfg

### ISSUE THREAD

**Issue by xrotwang** [fmatter/pylingdocs#4]
I was running `pylingdocs new` to create a new project, then `cd`-ed into the project and ran `pylingdocs preview` - which apparently picked up the project's `pylingdocs.cfg` successfully, but failed looking up the CLDF data. So I guess the path given for the `cldf` option in `pylingdocs.cfg` should be relative to the config file.

**Comment by fmatter** [comment]
I was not able to reproduce this.

**Comment by xrotwang** [comment]
```shell
$ pylingdocs new
INFO    Loading configuration
INFO    Loading templates
name [new-pld-document]: tt
...
$ cd tt
$ pylingdocs preview
INFO    Loading configuration
INFO    Loading templates
INFO    Rendering preview
ERROR   [Errno 2] No such file or directory: 'lgr-cldf/cldf/Generic-metadata.json'
ERROR   Could not load CLDF dataset from /home/robert_forkel/projects/pylingdocs/tt/lgr-cldf/cldf/Generic-metadata.json. Please specify a path to a valid CLDF metadata file.
```
Content of `pylingdocs.cfg` is
```ini
[PATHS]
cldf = lgr-cldf/cldf/Generic-metadata.json

[latex]
template = article
```
but would work with `cldf = ../lgr-cldf/cldf/Generic-metadata.json`, i.e. relative to the location of `pylingdocs.cfg`.

**Comment by fmatter** [comment]
I don't understand -- yes, the path to the cldf dataset is relative to the location of pylingdocs.cfg. What should be different?

**Comment by xrotwang** [comment]
It is **not** relative to `pylingdocs.cfg`, but relative to the directory in which `pylingdocs new` was run.

**Comment by xrotwang** [comment]
Thus, changing `cwd` between `pylingdocs new` and `pylingdocs preview` causes the error I'm seeing.

**Comment by xrotwang** [comment]
For context:
```shell
$ la
lgr-cldf
pylingdocs
tt
```

**Comment by fmatter** [comment]
> It is **not** relative to `pylingdocs.cfg`, but relative to the directory in which `pylingdocs new` was run.

OK, so is the issue that at the `pylingdocs new` stage you entered a path relative to the current directory, but then pylingdocs actually expects a path relative to the created project directory (`../bla/bla/bla.json`) ? If yes, what behavior would you expect / recommend? If no, I am still not understanding.

**Comment by xrotwang** [comment]
Well, at "pylingdocs new" I'm entering an existing path, because that's
supported via tab-completion in the shell. So this path is relative to cwd.
Since pylingdocs knows, where the cfg file will be created, I'd expect
pylingdocs to adapt the path I entered accordingly.

Florian Matter ***@***.***> schrieb am Mo., 19. Sep. 2022,
19:57:

**Comment by fmatter** [comment]
> Well, at "pylingdocs new" I'm entering an existing path, because that's supported via tab-completion in the shell. So this path is relative to cwd. Since pylingdocs knows, where the cfg file will be created, I'd expect pylingdocs to adapt the path I entered accordingly.

Right, I understand now! I don't have tab-completion in that context, interesting.

**Comment by xrotwang** [comment]
Ah, right, I don't have either. But I copy a path I got via tab-completion.
But maybe the better argument is that paths existing in cwd can be simpler
checked for existence - and would also be simpler to document in help.

Florian Matter ***@***.***> schrieb am Mo., 19. Sep. 2022,
20:37:
